src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi2/func_80A54C34.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi2/func_80A54C6C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi2/EnHeishi2_Draw.s")
+#endif
+
+void func_80A54954(EnHeishi2* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06005C30.genericHeader);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
+    this->actionFunc = func_80A549E8;
+}
+
+void func_80A549E8(EnHeishi2* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->unk_300 == func_8010BDBC(&globalCtx->msgCtx)) {
+        if (func_80106BC8(globalCtx) != 0) {
+            func_80106CCC(globalCtx);
+            if (this->initParams == 2) {
+                this->actionFunc = func_80A531E4;
+            }
+            if (this->initParams == 5) {
+                this->actionFunc = func_80A53908;
+            }
+        }
+    }
+}
+
+void EnHeishi2_Update(EnHeishi2* this, GlobalContext* globalCtx) {
+    ColliderCylinder* collider;
+    EnHeishi2* heishi2Temp;
+    Actor* thisx = &this->actor;
+    s32 phi_v1;
+
+    Actor_SetHeight(thisx, this->unk_2E0);
+    if ((this->initParams == 2) || (this->initParams == 5)) {
+        this->actor.posRot2.pos.y = 70.0f;
+        Actor_SetHeight(thisx, 70.0f);
+        func_80038290(globalCtx, thisx, &this->unk_260, &this->unk_26C, thisx->posRot2.pos);
+    }
+
+    this->unk_2FC++;
+    phi_v1 = 0;
+    heishi2Temp = this;
+
+    do {
+        phi_v1 += 2;
+        if (heishi2Temp->gateTimer != 0) {
+            heishi2Temp->gateTimer--;
+        }
+        heishi2Temp = (EnHeishi2*)&(heishi2Temp->actor.type); // what is going on here ???
+    } while (phi_v1 != 10);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(thisx);
+    if (this->initParams != 6) {
+        if (this->initParams != 9) {
+            func_8002E4B4(globalCtx, thisx, 10.0f, 10.0f, 30.0f, 0x1D);
+            collider = &this->collider;
+            Collider_CylinderUpdate(thisx, collider);
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, collider);
+        }
+    }
+}
+
+s32 EnHeishi2_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                               EnHeishi2* this) {
+    if ((this->initParams != 1) && (this->initParams != 7)) {
+        if (limbIndex == 9) {
+            rot->x = rot->x + this->unk_26C.y;
+        }
+        if (limbIndex == 16) {
+            rot->x = rot->x + this->unk_260.y;
+            rot->z = rot->z + this->unk_260.z;
+        }
+    }
+    return 0;
+}
+
+void EnHeishi2_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, EnHeishi2* this) {
+    if (limbIndex == 16) {
+        Matrix_Get(&this->mtxf_330);
+    }
+}
+
```

Comment:
both of these are going to take Actor* thisx and then you can use the macro to get access to your actor specific variable in override limb draw